src/data_provider/src/packages/appxWindowsWrapper.h

```diff
@@ -0,0 +1,387 @@
+                    }
+                }
+                catch (...)
+                {
+                }
+            }
+
+            const auto pos { name.find(INVALID_NAME_APP) };
+
+            /*
+             * If the name starts with "@{" string then we need to look for the name on the cache registry.
+             * The name obtained is the key for the app name.
+             */
+            if (pos != std::string::npos && pos == 0)
+            {
+                try
+                {
+                    name = searchNameFromCacheRegistry(fieldName, name);
+                }
+                catch (...)
+                {
+                }
+            }
+
+            const auto index { name.find(INVALID_NAME_APP) };
+            return (index != std::string::npos && index == 0) ? "" : name;
+        }
+
+        /*
+         * @brief Get application vendor name.
+         *
+         * @param[in] registry Registry object that points to where the sub-registries with the names of the registry with the data are located.
+         *
+         * @return Returns the vendor name
+         */
+        const std::string getVendor(Utils::Registry& registry)
+        {
+            std::string vendor;
+
+            for (const auto& folder : registry.enumerate())
+            {
+                try
+                {
+                    Utils::Registry fileReg(m_key, m_userId + "\\" + APPLICATION_STORE_REGISTRY + "\\" + m_nameApp + "\\" + folder + FILE_ASSOCIATIONS_REGISTRY, KEY_READ | KEY_QUERY_VALUE);
+                    vendor = searchPublisher(fileReg);
+
+                    if (vendor.size() == 0)
+                    {
+                        Utils::Registry urlReg(m_key, m_userId + "\\"  + APPLICATION_STORE_REGISTRY + "\\" + m_nameApp + "\\" + folder + URL_ASSOCIATIONS_REGISTRY, KEY_READ | KEY_QUERY_VALUE);
+                        vendor = searchPublisher(urlReg);
+                    }
+                }
+                catch (...)
+                {
+                }
+            }
+
+            return vendor;
+        }
+
+        /*
+         * @brief Search the name application in the cache registry.
+         *
+         * @param[in] nameApp Abbreviation of the name application to look for in the cache registry.
+         * @param[in] nameKey Name key where is the name application
+         *
+         * @return Return the name application.
+         */
+        const std::string searchNameFromCacheRegistry(const std::string& nameApp, const std::string& nameKey)
+        {
+            std::string registry;
+            std::string name;
+
+            // Looking for the named folder same like name app
+            for (const auto& folder : m_cacheReg)
+            {
+                if (folder.find(nameApp) != std::string::npos)
+                {
+                    registry = folder;
+                    break;
+                }
+            }
+
+            if (registry.size() != 0 )
```

Comment:
This should be changed anyway if `find_if` approach is used
```suggestion
            if (!registry.empty() )
```